docs/source/io/io.rst

```diff
@@ -0,0 +1,26 @@
+.. _io:
+
+Saving and Loading in Librascal
+===============================
+
+The library aims at being able to load/save three types of objects: atomic structures,
+their computed representation and trained machine learning models.
+Many tools already exist for reading atomic srtuctures in python and c++ so we rely on
+ASE_ on the python side and
+support the ASE `.json` atomic structure file format on the c++ side
+(:cpp:class:`AtomicStructure <rascal::AtomicStructure>`). An example is available
+in `json_structure.cc` and `spherical_invariants_example.cc`.
+At the moment we do not support saving representations in native Librascal format
```

Comment:
What about sparse points? It looked like you had an example for saving and loading those via JSON.

Reply:
sparse points are not a representation but basis points for the regression. And yes they can be saved to JSON format

Reply:
Ok, sure, but there's a way to save a _model_ (sparse points with representation hypers), which is what most users care about. I don't really see a use case for saving the representation class itself, if the calculated representation can be copied into a `SparsePoints` and saved that way.

If there is such a special use case, I would like to see separate documentation for it (since it seems like such a special case). Otherwise, I'm assuming there's a reason you didn't have to implement Representation serialization to get model serialization working for most use cases.

Reply:
> a way to save a model (sparse points with representation hypers)

the KRR model corresponds to sparse points, weights, kernel and representation. So each pieces can be saved on its own hence be saved in the bundle that is the KRR model.

> If there is such a special use case, I would like to see separate documentation for it (since it seems like such a special case). Otherwise, I'm assuming there's a reason you didn't have to implement Representation serialization to get model serialization working for most use cases.

I am really not following you here.

Reply:
Since the `get_features()` workaround was good enough for everything you tried -- along with perhaps the number of atoms per frame -- why do we need a "native librascal format" for representations? Just make the workaround official and that's it, until we find an application where it's not good enough.

Reply:
> why do we need a "native librascal format" for representations?

we have one since we have some dedicated dataholder in c++. It is just not convenient to serialize for the moment so we don't but in the not so near future (hopefully) we will have a simpler dataholders (that can be functional without a structure manager).
I would not recommend using `get_features` atm because it basically does not allow to used rascal on the features (model, kernel...).

Reply:
Now I'm more confused than ever. You're recommending against using `get_features()`, even though the next sentence in the docs acknowledges it as a workaround?

Ok, whatever, I'm willing to wait for the new dataholder type for an "official" way to save representations with all the necessary extra information.